### PR TITLE
fixes #3339 by documenting the limitations of case-statement

### LINKED ISSUE #3339: Implicit return with case expression fails with 'var' return.
``` nim
type
  Foo = ref object
    x: seq[string]

proc get_x(x: Foo): var seq[string] =
  case true
  of true:
    x.x
  else:
    x.x

var foo = Foo(x: @[])
foo.get_x().add("asd")
```

Error:

```
Error: expression has no address
```

Works when explicitly using `return`.

### PR DESCRIPTION
**Pull request by narimiran**

### CODE REVIEW

**Review by Varriount — APPROVED** [review]
Huh, I didn't know this. 
Should the old issue be kept open? I can see an argument for improving the semantics to allow `case` to return an lvalue

File: doc/manual.rst [hunk 1]
```diff
@@ -2843,6 +2843,35 @@ expanded into a list of its elements:
...
+won't work:
+
+.. code-block:: nim
+  type
+    Foo = ref object
+      x: seq[string]
+
+  proc get_x(x: Foo): var seq[string] =
+    # doesn't work
+    case true
```
Comment (kaushalmodi) [inline doc/manual.rst:2856]: Is `case true` even valid? It's like saying `case 123` for ints.

Comment (kaushalmodi) [reply doc/manual.rst:2856]: huh, I am surprised that this is valid: https://play.nim-lang.org/#ix=2b6d

Comment (Araq) [reply doc/manual.rst:2856]: why wouldn't it? bool is an ordinal types and case statements work on ordinals.

**Comment by Araq** [comment]
> Should the old issue be kept open? I can see an argument for improving the semantics to allow case to return an lvalue

It seems rather hard to support in the long run and ``try except`` and ``if elif else`` expressions are also affected. Let's simply say no to this one.

**Pull request merged**